servers/services/src/main/java/org/projectnessie/services/rest/NessieExceptionMapper.java

```diff
@@ -58,28 +62,39 @@ public NessieExceptionMapper(ServerConfig config) {
   public Response toResponse(Exception exception) {
     int status;
     String reason;
+    String message;
 
     if (exception instanceof WebApplicationException) {
       WebApplicationException e = (WebApplicationException) exception;
       Status st = Status.fromStatusCode(e.getResponse().getStatus());
       status = st.getStatusCode();
       reason = st.getReasonPhrase();
+      message = exception.getMessage();
     } else if (exception instanceof BaseNessieClientServerException) {
       // log message at debug level so we can review stack traces if enabled.
       LOGGER.debug("Exception on server with appropriate error sent to client.", exception);
       BaseNessieClientServerException e = (BaseNessieClientServerException) exception;
       status = e.getStatus();
       reason = e.getReason();
+      message = exception.getMessage();
     } else if (exception instanceof JsonParseException
         || exception instanceof JsonMappingException
         || exception instanceof IllegalArgumentException) {
       status = Status.BAD_REQUEST.getStatusCode();
       reason = Status.BAD_REQUEST.getReasonPhrase();
+      message = exception.getMessage();
+    } else if (exception instanceof BackendLimitExceededException) {
+      LOGGER.warn("Backend throttled/refused the request: {}", exception.toString());
```

Comment:
what is the  reasoning behind `debug` in some branches, `warn` in others and no log for `WebApplicationException`?

Reply:
TBH, I don't really know. For the `BackendLimitExceededException` the reason was to give an operator at least some visibility that some requests failed, because the backend's overloaded. We have debug-logging for all code-paths via `BaseExceptionMapper#buildExceptionResponse()`, but that one covers really all exceptions.

Reply:
But pushed another commit to prevent logging at debug twice